erpnext/selling/doctype/quotation/quotation.js

```diff
@@ -27,7 +27,11 @@ erpnext.selling.QuotationController = erpnext.selling.SellingController.extend({
 
 		var me = this;
 
-		if(doc.docstatus == 1 && doc.status!=='Lost') {
+		if (doc.__islocal) {
+			this.frm.set_value('valid_till', frappe.datetime.add_months(doc.transaction_date, 1))
+		}
+
+		if(doc.docstatus == 1 && doc.status!=='Lost' && (!doc.valid_till || doc.valid_till >= frappe.datetime.get_today())) {
```

Comment:
`doc.valid_till >= frappe.datetime.get_today()` it will compair strings !!

use the `frappe.datetime.get_diff` instead to check if valid_till is greater than or equal to get_today